sickbeard/providers/__init__.py

```diff
@@ -107,4 +106,5 @@ def getProviderClass(provider_id):
         for x in sickbeard.providerList + sickbeard.newznabProviderList + sickbeard.torrentRssProviderList
         if x
     }
-    return providers[provider_id]
+    if providers:
+        return providers.values()[0]
```

Comment:
Should be fixed now.
getProviderClass never returned a list, it always returned the first element of the list. So that's just the same, it's the first value of the dict. Since it makes very little sense to make a list/dict and return only the first element, I made it just directly return the first object now.